From 343fb8af47b63d2f572873280efc78ce5f2bb381 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 12 Dec 2017 15:59:41 -0800
Subject: [PATCH 02/10] dma: xilinx: dpdma: Use dma_pool_zalloc()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

drivers/dma/xilinx/xilinx_dpdma.c:713:9-10: WARNING: *_pool_zalloc
should be used for sw_desc, instead of *_pool_alloc/memset

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 1578b20af105de114165d28413ee6357496cc532)
---
 drivers/dma/xilinx/xilinx_dpdma.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_dpdma.c b/drivers/dma/xilinx/xilinx_dpdma.c
index 94833d0..c155dcd 100644
--- a/drivers/dma/xilinx/xilinx_dpdma.c
+++ b/drivers/dma/xilinx/xilinx_dpdma.c
@@ -713,11 +713,10 @@ static void xilinx_dpdma_sw_desc_addr_64(struct xilinx_dpdma_sw_desc *sw_desc,
 	struct xilinx_dpdma_sw_desc *sw_desc;
 	dma_addr_t phys;
 
-	sw_desc = dma_pool_alloc(chan->desc_pool, GFP_ATOMIC, &phys);
+	sw_desc = dma_pool_zalloc(chan->desc_pool, GFP_ATOMIC, &phys);
 	if (!sw_desc)
 		return NULL;
 
-	memset(sw_desc, 0, sizeof(*sw_desc));
 	sw_desc->phys = phys;
 
 	return sw_desc;
-- 
1.8.5.2.233.g932f7e4

